flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/data/conversion/ArrayObjectArrayConverter.java

```diff
@@ -154,7 +154,7 @@ void writeElement(int pos, E element) {
 
 	BinaryArrayData completeWriter() {
 		reuseWriter.complete();
-		return reuseArray;
+		return reuseArray.copy();
```

Comment:
I moved the `copy()` to the outer level in `toBinaryArrayData` because this method is shared with `MapMapConverter` which would perform a copy twice.